### PR TITLE
[GraphBolt] Rewrite DistributeItemSampler logic

### PR DESCRIPTION
**Pull request by RamonZhou**
## Description
Simplified the logic of DistributedItemSampler.
Added comments and updated the doc string as well as the test.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/utils/item_sampler_utils.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+          replicas.
+
+    Returns:
+        tuple: A tuple containing three numbers:
+            - start_offset (int): The starting offset of the range assigned to
+              the current worker.
+            - assigned_count (int): The length of the range assigned to the
+              current worker.
+            - evened_count (int): The number of items that the current worker
+              will produce after dropping uneven batches.
```
Comment (frozenbugs) [inline python/dgl/graphbolt/utils/item_sampler_utils.py:71]: if drop_uneven_inputs is false, do we still have evened_count?

File: python/dgl/graphbolt/item_sampler.py [hunk 2]
```diff
@@ -275,7 +214,7 @@ def __iter__(self):
                 if (
                     self._distributed
                     and self._drop_uneven_inputs
```
Comment (frozenbugs) [inline python/dgl/graphbolt/item_sampler.py:216]: Is it possible to also move this logic to calculate_range?

File: python/dgl/graphbolt/utils/item_sampler_utils.py (hunk 1 above)
Comment (RamonZhou) [reply python/dgl/graphbolt/utils/item_sampler_utils.py:71]: It will be the same as assigned_count.

File: python/dgl/graphbolt/item_sampler.py [hunk 3]
```diff
@@ -184,83 +184,18 @@ def __iter__(self):
...
-                        + self._batch_size
-                        * (
-                            total_batch_count // num_workers * worker_id
-                            + min(worker_id, total_batch_count % num_workers)
-                        )
-                        + last_batch // num_workers * worker_id
-                        + min(worker_id, last_batch % num_workers)
-                    )
+        total = len(self._item_set)
+        start_offset, num_items, num_output = calculate_range(
```
Comment (frozenbugs) [inline python/dgl/graphbolt/item_sampler.py:188]: Please align the internal variable and external variable name to avoid confusion.

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**